clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppQueue.kt

```diff
@@ -0,0 +1,65 @@
+package com.clevertap.android.sdk.inapp
+
+import androidx.annotation.WorkerThread
+import com.clevertap.android.sdk.CleverTapInstanceConfig
+import com.clevertap.android.sdk.Logger
+import com.clevertap.android.sdk.inapp.store.preference.StoreRegistry
+import org.json.JSONArray
+import org.json.JSONObject
+
+class InAppQueue(
+    private val config: CleverTapInstanceConfig,
+    private val storeRegistry: StoreRegistry
+) {
+
+    @WorkerThread
+    @Synchronized
+    fun enqueue(jsonObject: JSONObject) {
+        val currentQueue = getQueue()
+        currentQueue.put(jsonObject)
+        saveQueue(currentQueue)
+    }
+
+    @WorkerThread
+    @Synchronized
+    fun enqueueAll(jsonArray: JSONArray) {
+        val currentQueue = getQueue()
+        for (i in 0 until jsonArray.length()) {
+            try {
+                currentQueue.put(jsonArray.getJSONObject(i))
+            } catch (e: Exception) {
+                Logger.d(
+                    config.accountId,
+                    "InAppController: Malformed InApp notification: " + e.message
+                )
+            }
+        }
+        saveQueue(currentQueue)
+    }
+
+    @WorkerThread
+    @Synchronized
+    fun dequeue(): JSONObject? {
+        val currentQueue = getQueue()
+        if (currentQueue.length() == 0) {
+            return null
+        }
+        val removedObject = currentQueue.remove(0)
+        saveQueue(currentQueue)
+        return removedObject as JSONObject
```

Comment:
is this cast necessary here?

Reply:
forced cast can cause a crash, lets check instance or return null

Reply:
added changes in https://github.com/CleverTap/clevertap-android-sdk/commit/582552ec2b6eca94d1a939dacd0d1ef6ddaf36b5